components/backup/src/endpoint.rs

```diff
@@ -660,87 +702,30 @@ impl<E: Engine, R: RegionInfoProvider> Endpoint<E, R> {
         for _ in 0..concurrency {
-            self.spawn_backup_worker(prs.clone(), request.clone(), res_tx.clone());
-        }
-
-        // Drop the extra sender so that for loop does not hang up.
-        drop(res_tx);
-        let mut summary = Statistics::default();
-        for (brange, res) in res_rx {
-            let start_key = if request.is_raw_kv {
-                brange
-                    .start_key
-                    .map_or_else(|| vec![], |k| k.into_encoded())
-            } else {
-                brange
-                    .start_key
-                    .map_or_else(|| vec![], |k| k.into_raw().unwrap())
-            };
-            let end_key = if request.is_raw_kv {
-                brange.end_key.map_or_else(|| vec![], |k| k.into_encoded())
-            } else {
-                brange
-                    .end_key
-                    .map_or_else(|| vec![], |k| k.into_raw().unwrap())
-            };
-            let mut response = BackupResponse::default();
-            match res {
-                Ok((mut files, stat)) => {
-                    debug!("backup region finish";
-                        "region" => ?brange.region,
-                        "start_key" => hex::encode_upper(&start_key),
-                        "end_key" => hex::encode_upper(&end_key),
-                        "details" => ?stat);
-                    summary.add(&stat);
-                    // Fill key range and ts.
-                    for file in files.iter_mut() {
-                        file.set_start_key(start_key.clone());
-                        file.set_end_key(end_key.clone());
-                        file.set_start_version(request.start_ts.into_inner());
-                        file.set_end_version(request.end_ts.into_inner());
-                    }
-                    response.set_files(files.into());
-                }
-                Err(e) => {
-                    error!("backup region failed";
-                        "region" => ?brange.region,
-                        "start_key" => hex::encode_upper(response.get_start_key()),
-                        "end_key" => hex::encode_upper(response.get_end_key()),
-                        "error" => ?e);
-                    response.set_error(e.into());
-                }
-            }
-            response.set_start_key(start_key);
-            response.set_end_key(end_key);
-            if let Err(e) = resp.unbounded_send(response) {
-                error!("backup failed to send response"; "error" => ?e);
-                break;
-            }
+            self.spawn_backup_worker(prs.clone(), request.clone(), resp.clone());
         }
         let duration = start.elapsed();
         BACKUP_REQUEST_HISTOGRAM.observe(duration.as_secs_f64());
-        info!("backup finished";
-            "take" => ?duration,
-            "summary" => ?summary);
+        info!("backup_ts finished";
+            "take" => ?duration)
```

Comment:
Backup is running in workers it may not finish yet.

Reply:
so, let's move this log?